hydra-main/src/main/java/com/addthis/hydra/job/web/SpawnService.java

```diff
@@ -39,93 +39,125 @@
-            jetty.setConnectors(new Connector[]{selectChannelConnector});
-            sslEnabled = false;
+    public void start() throws Exception {
+        PrometheusServletCreator.register();
+        server.start();
+    }
+
+    private Server createServer(final Spawn spawn, SpawnServiceConfiguration configuration) throws Exception {
+        Server server = new Server();
+
+        initServerConnectors(server, spawn, configuration);
+
+        // for closing certain api resource, namely JobsResource, but new ones can be added
+        final Closer closer = Closer.create();
+        ServletContextHandler servletContextHandler = createServletContextHandler(spawn, configuration, closer);
+        Handler handler = createRootHandler(servletContextHandler);
+
+        server.setAttribute("org.eclipse.jetty.Request.maxFormContentSize", 5000000);
+        server.setHandler(handler);
+
+        // this must be after server.setHandler(handler) because websocket configuration needs
+        // the server object from servletContextHandler
+        configureWebSocketServlet(servletContextHandler);
+
+        server.addLifeCycleListener(new AbstractLifeCycle.AbstractLifeCycleListener() {
+            @Override
+            public void lifeCycleStopping(LifeCycle event) {
+                super.lifeCycleStopping(event);
+                try {
+                    closer.close();
+                } catch (IOException ex) {
+                    log.error("IOException while closing jetty resources: ", ex);
+                }
+            }
+        });
+        return server;
+    }
+
+    private void initServerConnectors(Server server, Spawn spawn, SpawnServiceConfiguration configuration)
+            throws IOException {
+        // http
+        ServerConnector http = new ServerConnector(server);
+        http.setPort(configuration.webPort);
+        server.addConnector(http);
+
+        // https
+        if (configuration.requireSSL) {
+            SslConnectionFactory sslConnectionFactory = createSslConnectionFactory(configuration);
+            ServerConnector https = new ServerConnector(server, sslConnectionFactory, new HttpConnectionFactory());
+            https.setPort(configuration.webPortSSL);
+            server.addConnector(https);
+            spawn.getSystemManager().updateSslEnabled(true);
         }
-        spawn.getSystemManager().updateSslEnabled(sslEnabled);
+    }
+
+    private SslConnectionFactory createSslConnectionFactory(SpawnServiceConfiguration configuration)
+            throws IOException {
+        SslContextFactory sslContextFactory = new SslContextFactory();
+        sslContextFactory.setKeyStorePath(configuration.keyStorePath);
+        sslContextFactory.setKeyStorePassword(readFile(configuration.keyStorePassword));
+        sslContextFactory.setKeyManagerPassword(readFile(configuration.keyManagerPassword));
+        return new SslConnectionFactory(sslContextFactory, HttpVersion.HTTP_1_1.asString());
```

Comment:
Can you do http2?

Reply:
Jetty has http2 support, but I don't know much of it and what that entails for hydra

Reply:
It should be invisible for Hydra if you enable it.

Reply:
Turns out that enabling HTTP 2 on the server side is non-trivial amount of work, especially for TLS:

http://www.eclipse.org/jetty/documentation/9.4.8.v20171121/http2.html

Going to leave it as a potential future improvement for now.